spec/system/admin/reports/enterprise_summary_fees/enterprise_summary_fee_with_tax_report_by_producer_spec.rb

```diff
@@ -31,54 +31,87 @@
 
-  let!(:coordinator_fees){
+  let(:coordinator_fees){
     create(:enterprise_fee, :flat_rate, enterprise: distributor, amount: 20,
                                         name: 'Adminstration',
                                         fee_type: 'admin',
                                         tax_category: tax_category)
   }
-  let!(:supplier_fees){
-    create(:enterprise_fee, :flat_rate, enterprise: supplier, amount: 15,
-                                        name: 'Transport',
-                                        fee_type: 'transport',
-                                        tax_category: tax_category)
+  let(:supplier_fees){
+    create(:enterprise_fee, :per_item, enterprise: supplier, amount: 15,
+                                       name: 'Transport',
+                                       fee_type: 'transport',
+                                       tax_category: tax_category)
   }
-  let!(:supplier_fees2){
-    create(:enterprise_fee, :flat_rate, enterprise: supplier2, amount: 25,
-                                        name: 'Sales',
-                                        fee_type: 'sales',
-                                        tax_category: tax_category)
+  let(:supplier_fees2){
+    create(:enterprise_fee, :per_item, enterprise: supplier2, amount: 25,
+                                       name: 'Sales',
+                                       fee_type: 'sales',
+                                       tax_category: tax_category)
   }
-  let!(:distributor_fee){
+  let(:distributor_fee){
     create(:enterprise_fee, :flat_rate, enterprise: distributor, amount: 10,
                                         name: 'Packing',
                                         fee_type: 'packing',
                                         tax_category: tax_category)
   }
 
+  let!(:incoming_exchange1) {
+    order_cycle.exchanges.create! sender: supplier, receiver: distributor, incoming: true
+  }
+
+  let!(:incoming_exchange2) {
+    order_cycle.exchanges.create! sender: supplier2, receiver: distributor, incoming: true
+  }
+  let(:outgoing_exchange) {
+    order_cycle.exchanges.create! sender: distributor, receiver: distributor, incoming: false
+  }
+
+  let!(:order){ create(:order_with_distributor, distributor: distributor) }
   before do
     order_cycle.coordinator_fees << coordinator_fees
     order_cycle.exchanges.incoming.first.exchange_fees.create!(enterprise_fee: supplier_fees)
+    order_cycle.exchanges.incoming.first.exchange_variants.create!(variant: variant)
     order_cycle.exchanges.incoming.second.exchange_fees.create!(enterprise_fee: supplier_fees2)
+    order_cycle.exchanges.incoming.second.exchange_variants.create!(variant: variant2)
     order_cycle.exchanges.outgoing.first.exchange_fees.create!(enterprise_fee: distributor_fee)
+    order_cycle.exchanges.outgoing.first.exchange_variants.create!(variant: variant)
+    order_cycle.exchanges.outgoing.first.exchange_variants.create!(variant: variant2)
```

Comment:
Ping @dacook, from our chat earlier on: this setup works correctly, as opposed to the changes I've introduced on the [previous commit](https://github.com/openfoodfoundation/openfoodnetwork/pull/10797/commits/bd8e7404192aebc75326165163a3792095eaf743#r1185335232).